### PR TITLE
[Improve][Connector[File] Optimize files commit order

### PR DESCRIPTION
**Pull request by hailin0**
Before using `HashMap` store files path, so every checkpoint file commit is out of order.

Now switch to using `LinkedHashMap` to ensure that files are commit in the generated order

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileAggregatedCommitInfo.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class FileAggregatedCommitInfo implements Serializable {
      *
      * <p>V is the target file path of the data file.
      */
-    private final Map<String, Map<String, String>> transactionMap;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileAggregatedCommitInfo.java:37]: Why change it?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileCommitInfo.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public class FileCommitInfo implements Serializable {
      *
      * <p>V is the target file path of the data file.
      */
-    private final Map<String, String> needMoveFiles;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileCommitInfo.java:37]: The same as above.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileSinkAggregatedCommitter.java [hunk 3]
```diff
@@ -44,7 +44,7 @@ public List<FileAggregatedCommitInfo> commit(
         aggregatedCommitInfos.forEach(
                 aggregatedCommitInfo -> {
                     try {
-                        for (Map.Entry<String, Map<String, String>> entry :
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileSinkAggregatedCommitter.java:47]: The same as above.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/state/FileSinkState.java [hunk 4]
```diff
@@ -21,16 +21,16 @@
...
 import java.util.List;
-import java.util.Map;
 
 @Data
 @AllArgsConstructor
 public class FileSinkState implements Serializable {
     private final String transactionId;
     private final String uuidPrefix;
     private final Long checkpointId;
-    private final Map<String, String> needMoveFiles;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/state/FileSinkState.java:33]: The same as above.

**Review by TyrantLucifer — COMMENTED** [review]
I think keep the Top-level interface definition is a good choice.

**Comment by hailin0** [comment]
> I think keep the Top-level interface definition is a good choice.

@TyrantLucifer 
I want to explicitly emphasize this feature at the top level interface

Otherwise, the internal implementation may be modified by someone at any time to lose this feature

**Comment by TyrantLucifer** [comment]
> > I think keep the Top-level interface definition is a good choice.
> 
> @TyrantLucifer I want to explicitly emphasize this feature at the top level interface
> 
> Otherwise, the internal implementation may be modified by someone at any time to lose this feature

IMO, Emphasize certain rules that do not necessarily need to be reflected in the code definition. Annotations can be written to provide developers with hints. The design principles of the code should not be broken by actual business rules.

**Comment by hailin0** [comment]
> > > I think keep the Top-level interface definition is a good choice.
> > 
> > 
> > @TyrantLucifer I want to explicitly emphasize this feature at the top level interface
> > Otherwise, the internal implementation may be modified by someone at any time to lose this feature
> 
> IMO, Emphasize certain rules that do not necessarily need to be reflected in the code definition. Annotations can be written to provide developers with hints. The design principles of the code should not be broken by actual business rules.

With the current review quality, interface restrictions are less likely to be broken, mainly because interface restrictions currently have no side effects

**Review by TyrantLucifer — APPROVED** [review]

**Comment by hailin0** [comment]
@EricJoy2048 @Hisoka-X @ic4y PTAL

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**